### PR TITLE
Allow installing Python bindings with pip

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

This PR replaces the script that appends the Python modules to the `PYTHONPATH` with functionality that allows installing them with `pip`. The modules can be installed into a Python environment in _development mode_, which means they are symlinked so that changes to the modules will be reflected in the Python environment, with the command `pip install -e path/to/bin/python`.

For those working on Python code for SpECTRE, I would like to note the following (unrelated to this PR):

Once the modules are installed in the Python environment, they can also be run directly from the command line with the command `python -m the_module [args]`. This runs a `__main__.py` script in the package directory or the specific module file as `__main__`. See e.g. https://docs.python.org/3.7/library/__main__.html. We can use this functionality to write Python modules that both provide functions to use e.g. in a Jupyter notebook _and_ expose a CLI, e.g. to make visualisations.

### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsdeppe** [comment]
I'm fine with adding this functionality, but I think we should keep supporting the `PYTHONPATH` method. Does this PR also support conda by chance? If so, would you mind stating something about that?

**Comment by nilsvu** [comment]
Ok sounds good, I added the `PYTHONPATH` method back in.

With Anaconda you also install local packages with `pip`, so it works the same way. I added a note to the documentation.

File: src/PythonBindings/setup.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from distutils.core import setup
+
+setup(
+    name='spectre',
+    version='0.1',
+    description="Python bindings for the SpECTRE code",
```
Comment (nilsdeppe) [inline src/PythonBindings/setup.py]: `for the code SpECTRE` or just `for SpECTRE`

**Review by nilsdeppe — COMMENTED** [review]
One very small comment. I like this. Thanks for the conda clarification! That'll be helpful for us python non-experts :)

Edit: I'll explicitly state I'm happy for you to squash this trivial comment :)

File: src/PythonBindings/setup.py (hunk 1 above)
Comment (nilsvu) [reply src/PythonBindings/setup.py]: Fixed this. I should note that we'll certainly want to extend the metadata in this file in the future, e.g. add a `long_description`, do proper versioning, add contact email addresses, and so on.

Comment (nilsdeppe) [reply src/PythonBindings/setup.py]: Ah okay, cool! Once we get away from version 0 I guess we can set that according to the SpECTRE version?

**Review by nilsdeppe — DISMISSED** [review]

File: src/PythonBindings/setup.py (hunk 1 above)
Comment (nilsvu) [reply src/PythonBindings/setup.py]: Yeah, we can do that with a CMake replace.

File: src/PythonBindings/setup.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env python
+
+# Distributed under the MIT License.
+# See LICENSE.txt for details.
+
+from distutils.core import setup
+
+setup(
+    name='spectre',
+    version='0.1',
```
Comment (kidder) [inline src/PythonBindings/setup.py]: what does version refer to?  If it is the version of spectre then it should be not be hard-coded.

Comment (nilsvu) [reply src/PythonBindings/setup.py]: It is the version of the Python package. I have put no thought into how to version this yet. Is there a CMake variable I can substitute in here? Or do you have another suggestion?

Comment (nilsvu) [reply src/PythonBindings/setup.py]: I don’t think we have even decided how to do versioning.. #615

Comment (nilsvu) [reply src/PythonBindings/setup.py]: I just substitute in the `SpECTRE_VERSION` here for now. Seems reasonable for the Python bindings to share the main code's version.

**Review by kidder — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**